src/sentry/api/endpoints/organization_events_spans_performance.py

```diff
@@ -135,23 +155,69 @@ def data_fn(offset: int, limit: int) -> Any:
                 max_per_page=4,
             )
 
-    def get_orderby_column(self, request: Request) -> Tuple[str, str]:
-        orderbys = super().get_orderby(request)
 
-        if orderbys is None:
-            direction = "-"
-            orderby = "sumExclusiveTime"
-        elif len(orderbys) != 1:
-            raise ParseError(detail="Can only order by one column.")
-        else:
-            direction = "-" if orderbys[0].startswith("-") else ""
-            orderby = orderbys[0].lstrip("-")
+class SpansExamplesSerializer(serializers.Serializer):  # type: ignore
+    query = serializers.CharField(required=False, allow_null=True)
+    span = ListField(child=serializers.CharField(), required=True, allow_null=False, max_length=10)
 
-        if orderby not in SPAN_PERFORMANCE_COLUMNS:
-            options = ", ".join(SPAN_PERFORMANCE_COLUMNS.keys())
-            raise ParseError(detail=f"Can only order by one of {options}")
+    def validate_span(self, spans: List[str]) -> List[Span]:
+        try:
+            return [Span.from_str(span) for span in spans]
+        except ValueError as e:
+            raise serializers.ValidationError(str(e))
 
-        return direction, orderby
+
+class OrganizationEventsSpansEndpoint(OrganizationEventsSpansEndpointBase):
+    def get(self, request: Request, organization: Organization) -> Response:
+        if not self.has_feature(request, organization):
+            return Response(status=404)
+
+        try:
+            params = self.get_snuba_params(request, organization)
+        except NoProjects:
+            return Response(status=404)
+
+        serializer = SpansExamplesSerializer(data=request.GET)
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=400)
+        serialized = serializer.validated_data
+
+        query = serialized.get("query")
+        spans = serialized["span"]
+
+        direction, orderby_column = self.get_orderby_column(request)
+
+        def data_fn(offset: int, limit: int) -> Any:
```

Comment:
Just checking, are we not able to type this response using a TypeDict?

Reply:
That's a good point, I think we can, I'll do this in a follow-up to improve the typing more.